client/trino-client/src/main/java/io/trino/client/spooling/encoding/CompressedQueryDataDecoder.java

```diff
@@ -52,7 +51,7 @@ public ResultRows decode(InputStream stream, DataAttributes metadata)
             try (InputStream inputStream = stream) {
                 byte[] input = new byte[segmentSize];
                 byte[] output = new byte[uncompressedSize];
-                int readBytes = ByteStreams.read(inputStream, input, 0, segmentSize);
+                int readBytes = inputStream.read(input, 0, segmentSize);
```

Comment:
But these are not equivalent - `ByteStreams#read` will read as much as possible, while `inputStream.read` can do a short read. I think you want `inputStream.readNBytes`.